incremental-font-transfer/src/ift_graph.rs

```diff
@@ -0,0 +1,261 @@
+
+    let mut result: BTreeSet<Tag> = Default::default();
+
+    if let Ok(gsub) = font.gsub() {
+        for fr in gsub.feature_list()?.feature_records() {
+            if standard_features.contains(&fr.feature_tag()) {
+                continue;
+            }
+
+            result.insert(fr.feature_tag());
+        }
+    }
+
+    if let Ok(gpos) = font.gpos() {
+        for fr in gpos.feature_list()?.feature_records() {
+            if standard_features.contains(&fr.feature_tag()) {
+                continue;
+            }
+
+            result.insert(fr.feature_tag());
+        }
+    }
+
+    Ok(result)
+}
+
+fn get_node_name(font: &FontRef<'_>) -> Result<String, ReadError> {
+    let chars: BTreeSet<char> = font
+        .charmap()
+        .mappings()
+        .map(|(cp, _)| char::from_u32(cp).unwrap())
+        .collect();
+
+    let mut name: String = chars.into_iter().collect();
+
+    let features = get_feature_tags(font)?;
+    if !features.is_empty() {
+        let features: Vec<_> = features.into_iter().map(|t| t.to_string()).collect();
+        let features = features.join(",");
+        name.push('|');
+        name.push_str(&features);
+    }
+
+    let axes = font.axes();
+    if !axes.is_empty() {
+        let axis_strings: Vec<_> = font
+            .axes()
+            .iter()
+            .map(|axis| {
+                axis.tag().to_string()
+                    + "["
+                    + &axis.min_value().to_string()
+                    + ".."
+                    + &axis.max_value().to_string()
+                    + "]"
+            })
+            .collect();
+        let axis_strings = axis_strings.join(",");
+        name.push('|');
+        name.push_str(&axis_strings);
+    }
+
+    Ok(name)
+}
+
+fn to_next_font(
+    base_path: &Path,
+    font: &FontRef<'_>,
+    patch_uri: PatchUri,
+) -> Result<Vec<u8>, UriTemplateError> {
+    let path = base_path.join(patch_uri.uri_string()?);
+    let patch_bytes = std::fs::read(path.clone())
+        .unwrap_or_else(|e| panic!("Unable to read patch file ({}): {:?}", path.display(), e));
+
+    let patch_info: PatchInfo = patch_uri.try_into()?;
+
+    Ok(font
+        .apply_table_keyed_patch(&patch_info, &patch_bytes)
+        .expect("Patch application failed."))
+}
+
+fn to_graph(
+    base_path: &Path,
+    font: FontRef<'_>,
+    mut graph: BTreeMap<String, BTreeSet<String>>,
+) -> Result<BTreeMap<String, BTreeSet<String>>, UriTemplateError> {
+    let patches =
+        intersecting_patches(&font, &SubsetDefinition::all()).expect("patch map parsing failed");
+
+    let node_name = get_node_name(&font).unwrap();
+    graph.entry(node_name.clone()).or_default();
+
+    for patch in patches {
+        match patch.encoding() {
+            PatchFormat::TableKeyed { .. } => {}
+            // This graph only considers invalidating (that is table keyed patches), so skip all other types.
+            _ => continue,
+        };
```

Comment:
I'd probably write this as `if !matches!(patch.encoding(), PatchFormat::TableKeyed { .. }) { .. }` but this is totally fine 🤷